### PR TITLE
Subspace Templates

### PR DESCRIPTION
**Pull request by techsmyth**
Usage of TemplatesManager on Space entity to replace
- spaceDefaults
- library (templatesSet)

Updated mutation to be for TemplateDefault.

To do:
- translation file keys + text
- ...

## Release Notes

- **New Features**
	- Introduced collaboration templates with enhanced data handling and UI components.
	- Added new dialogs for selecting and applying collaboration templates.
	- Enhanced error handling and loading states across various components.
	- Expanded GraphQL queries to include collaboration-related fields and mutations.
	- Implemented a new URL parser for better handling of collaboration space URLs.

- **Bug Fixes**
	- Improved validation and error messaging for URL inputs and template creation.

- **Documentation**
	- Updated documentation to reflect new collaboration template functionalities and usage.

- **Refactor**
	- Reorganized data structures and component logic to better support collaboration templates.

- **Style**
	- Enhanced UI components for better user experience and accessibility.

- **Tests**
	- Added tests for new components and functionalities related to collaboration templates.

### CODE REVIEW

**Comment by ccanos** [comment]
@coderabbitai review

**Review by bobbykolev — APPROVED** [review]
Tested locally, working well.
Some polishing was applied.
I think we're good to go!
Good job @techsmyth and @ccanos 👏

File: src/domain/journey/space/pages/SpaceSubspaces/SubspaceListView.tsx [hunk 1]
```diff
@@ -141,45 +154,39 @@ export const SubspaceListView: FC = () => {
...
+    setSaveAsTemplateDialogSelectedItem(undefined);
   };
 
   const onDeleteConfirmation = async () => {
-    if (selectedItem) {
-      await handleDelete(selectedItem);
-      setDeleteDialogOpen(false);
+    if (deleteDialogSelectedItem) {
+      await handleDelete(deleteDialogSelectedItem);
+      setDeleteDialogSelectedItem(undefined);
```
[the remark this answers is not in the document]
Comment (bobbykolev) [reply src/domain/journey/space/pages/SpaceSubspaces/SubspaceListView.tsx:173]: Please, remember that we have a global error handler, and we don't need to try and catch errors locally. Please, limit such reports in the future.

**Comment by bobbykolev** [comment]
@coderabbitai pause

**Comment by bobbykolev** [comment]
@coderabbitai resolve

**Review by ccanos — APPROVED** [review]

**Pull request merged**